CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+
+## Building Locally
+
+These are general steps that all modules in `libs/` adhere to give the prerequisite of the installation of `Rye`.
+
+1. Run `rye sync` in the module folder you want to work on (EG `rye sync --all-features` to work on `labelbox`).
+2. Run `rye build` to create a distribution in `dist/` which you can install into a Python environment (EG `pip install dist/labelbox-build.tar.gz`).
+
+It is generally **not** recommended to do `pip install -e .` with any Labelbox module to avoid virtual environment pollution. If you really want to modify the SDK while making it compatible with your existing `pip` based projects, use the method listed above.
+
+## Testing
+
+Each module within the repository will generally have three components to testing: unit testing, integration testing, and linting/formatting.
+
+### Unit Testing
+
+```bash
+rye run unit
+```
+
+### Integration Testing
+
+```bash
+LABELBOX_TEST_API_KEY="YOUR_API_TEST_KEY" LABELBOX_TEST_ENVIRON="prod" rye run integration
+```
+For more info on how to get a `LABELBOX_TEST_API_KEY` [Labelbox API key docs](https://labelbox.helpdocs.io/docs/api/getting-started). 
+
+**Integration tests by default will run against your account that you provide an API Key from and modify its data. If you want to run integration tests, without it impacting your existing account, create an additional account using a secondary e-mail on [Labelbox](https://labelbox.com). Free accounts are sufficent for integration testing purposes.**
+
+You can also use a `.env` file if you prefer instead of needing to type out the environmental overrides every single time you want to run commands. Please add a `--env-file` parameter to the test command (EG `rye --env-file=.env run integration`).
+
+### (Optional) Data Testing
+
+For testing the impact of the extra installs included with `labelbox[data]`, run the following:
+
+```bash
+LABELBOX_TEST_API_KEY="YOUR_API_TEST_KEY" LABELBOX_TEST_ENVIRON="prod" rye run data
+```
+
+By default `rye sync` does not install the extra packages needed in `labelbox[data]`. You'll need to run `rye sync --all-features`. Do not checkin `requirements.lock` or `requirements-dev.lock` after doing this.
+
+### Linting / Formatting
+
+Before making a commit, to automatically adhere to our formatting standards, it is recommended to install and activate [pre-commit](https://pre-commit.com/)
+```shell
+pip install pre-commit
+pre-commit install
+```
+After the above, running `git commit ...` will attempt to fix formatting,
+and make necessary changes to files. You will then need to stage those files again.
+
+You may also manually format your code by running the following:
+```bash
+rye run lint
+```
+
+### Documentation
```

Comment:
I wonder what are the changes to our sdk release process documented here https://labelbox.atlassian.net/wiki/spaces/PLT/pages/1721139216/Release+a+new+Python+SDK+Version#Step-4%3A-Update-readthedocs-and-verify-package-in-pypi but also elsewhere in the doc